pkg/security/security_profile/ad.go

```diff
@@ -557,56 +622,195 @@ func (m *Manager) HandleCGroupTracingEvent(event *model.CgroupTracingEvent) {
+func (m *Manager) SyncTracedCgroups() {
+	m.m.Lock()
+	defer m.m.Unlock()
+	m.syncTracedCgroups()
+}
+
+// manager should be locked
+func (m *Manager) syncTracedCgroups() {
 	if !m.config.RuntimeSecurity.ActivityDumpEnabled {
 		return
 	}
 
 	var err error
 	var event model.CgroupTracingEvent
-	var cgroupFile model.PathKey
+	var cgroupInode uint64
 	iterator := m.tracedCgroupsMap.Iterate()
 	seclog.Infof("snapshotting traced_cgroups map")
 
-	for iterator.Next(&cgroupFile, &event.ConfigCookie) {
-		m.m.Lock()
-		if m.ignoreFromSnapshot[cgroupFile] {
-			m.m.Unlock()
+	// Collect cgroups to delete during iteration, execute deletion AFTER
+	// to avoid modifying the map while iterating (causes "iteration aborted")
+	var cgroupsToDelete []uint64
+	for iterator.Next(&cgroupInode, &event.ConfigCookie) {
+		if m.ignoreFromSnapshot[cgroupInode] {
+			// mark for deletion - will delete AFTER iteration
+			cgroupsToDelete = append(cgroupsToDelete, cgroupInode)
 			continue
 		}
-		m.m.Unlock()
 
 		if err = m.activityDumpsConfigMap.Lookup(&event.ConfigCookie, &event.Config); err != nil {
-			// this config doesn't exist anymore, remove expired entries
-			seclog.Warnf("config not found for (%v): %v", cgroupFile, err)
-			_ = m.tracedCgroupsMap.Delete(cgroupFile)
+			// this config doesn't exist anymore, mark for deletion
+			seclog.Warnf("config not found for inode (%v): %v", cgroupInode, err)
+			cgroupsToDelete = append(cgroupsToDelete, cgroupInode)
 			continue
 		}
 
-		cgroupContext, _, err := m.resolvers.ResolveCGroupContext(cgroupFile)
+		cgroupContext, _, err := m.resolvers.ResolveCGroupContext(model.PathKey{Inode: cgroupInode})
 		if err != nil {
-			seclog.Warnf("couldn't resolve cgroup context for (%v): %v", cgroupFile, err)
-			_ = m.tracedCgroupsMap.Delete(cgroupFile)
+			seclog.Warnf("couldn't resolve cgroup context for inode (%v): %v", cgroupInode, err)
+			cgroupsToDelete = append(cgroupsToDelete, cgroupInode)
 			continue
 		}
+
+		// Just validate that this cgroup corresponds to an existing dump, don't create new ones
+		// We only keep the cgroup in the kernel map if it has an active dump
 		event.CGroupContext = *cgroupContext
 		event.ContainerContext.ContainerID = containerutils.FindContainerID(event.CGroupContext.CGroupID)
 
-		m.HandleCGroupTracingEvent(&event)
+		hasActiveDump := false
+		for _, ad := range m.activeDumps {
+			if ad.Profile.Metadata.ContainerID == event.ContainerContext.ContainerID {
+				hasActiveDump = true
+				break
+			}
+		}
+
+		if !hasActiveDump {
+			// No active dump for this cgroup - mark for deletion
+			cgroupsToDelete = append(cgroupsToDelete, cgroupInode)
+		}
 	}
 
 	if err = iterator.Err(); err != nil {
 		seclog.Warnf("couldn't iterate over the map traced_cgroups: %v", err)
 	}
+
+	// Delete all marked cgroups AFTER iteration (safe to modify map now)
+	for _, cgroupInode := range cgroupsToDelete {
+		_ = m.tracedCgroupsMap.Delete(cgroupInode)
+	}
+}
+
+// Used for tests only
```

Comment:
There's a [`security_profile/testing.go`](https://github.com/DataDog/datadog-agent/blob/cf6b9b734b646d6c6d83233bc177daa0d16bcae3/pkg/security/security_profile/testing.go) file for test-specific methods if you want to add these in there